torch/_inductor/codegen/triton.py

```diff
@@ -1008,6 +1008,49 @@ def mask_loads(self, mask):
         finally:
             self._load_mask = prior
 
+    def gen_assert_indirect_indexing(self, buffer, original_index, mask):
+        if mask == "None":
+            return
+        body = self.current_node._body
+        indirect_size = dict(zip(body.indirect_vars, body.indirect_max_sizes))
+        indirect_name = body.indirect_new
+        # Many indirect variables may be mapped to the same CSE'd variable
+        var_size = {}
+        for ind, size in indirect_size.items():
+            var_size.setdefault(indirect_name[ind], []).append(
+                V.kernel.rename_indexing(size)
+            )
+
+        indirect_vars = [
+            s for s in original_index.free_symbols if s.name.startswith("tmp")
+        ]
+        for var in indirect_vars:
+            sizes = var_size[var]
+            if all(isinstance(s, sympy.Integer) for s in sizes):
```

Comment:
how can one var correspond to different sizes?

Reply:
When you do `x[y, y]` with `torch.randn(3, 2)`, for example. I'll leave a comment.